packages/loopring_v3/contracts/impl/agents/FastWithdrawalAgent.sol

```diff
@@ -0,0 +1,203 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+pragma experimental ABIEncoderV2;
+
+import "../../iface/IExchangeV3.sol";
+
+import "../../lib/Claimable.sol";
+import "../../lib/ERC20SafeTransfer.sol";
+import "../../lib/MathUint.sol";
+import "../../lib/ReentrancyGuard.sol";
+import "../../lib/SignatureUtil.sol";
+import "../../lib/EIP712.sol";
+
+
+/// @title Fast withdrawal agent implementation.
+///
+///        A mediator/liquidity provider (whoever provides the funds immediately) can call
+///        `executeFastWithdrawals` to provide users immediately with funds onchain.
+///        This allows the security of the funds to be handled by any EOA or smart contract.
+///
+///        Users that want to make use of this functionality have to
+///        authorize this contract as their agent.
+///
+///        TODO: generalize this (here or another agent) with any function call
+///        a user wants to do to be able to simulate completely onchain smart wallet with
+///        funds stored in the Loopring protocol.
+///
+/// @author Brecht Devos - <brecht@loopring.org>
+contract FastWithdrawalAgent is ReentrancyGuard
+{
+    using AddressUtil       for address;
+    using AddressUtil       for address payable;
+    using ERC20SafeTransfer for address;
+    using MathUint          for uint;
+    using SignatureUtil     for bytes32;
+
+    struct FastWithdrawal
+    {
+        address exchange;
+        address from;
```

Comment:
is `from` the mediator? 
or `from` is the user who wants to withdrawal, and `to` can be either the user or another recipient address.

Reply:
The 2nd one. `from` is the user who wants to withdraw, `to` is the recipient address.

Same reasoning as for the deposit/transfer parameters above: hiding the account system as much as possible for users.

Reply:
Cool